### ISSUE TITLE
[Bug]: App crashes after typing the "ˋ" character

### ISSUE THREAD

**Issue by IzanRodrigo** [AndroidIDEOfficial/AndroidIDE#1495]
### What happened?

I was editing a file and typed the "ˋ" key and the app crashed.

### What's the expected behavior?

The app should not crash.

### What version of AndroidIDE you're using?

v2.6.0

### Relevant log output

```
AndroidIDE Crash Report
Version : v2.6.0-beta (260)
CI Build : true
Branch : main
Commit : 34d7e34e4
Variant : arm64-v8a (release)
SDK Version : 33
Supported ABIs : [arm64-v8a, armeabi-v7a, armeabi]
Manufacturer : Xiaomi
Device : 23043RP34G

Stacktrace:
java.lang.IllegalArgumentException: Not a valid Unicode code point: 0x800002CB
	at java.lang.Character.toChars(Character.java:9280)
	at io.github.rosemoe.sora.widget.CodeEditor.onKeyDown(Unknown Source:1238)
	at android.view.KeyEvent.dispatch(KeyEvent.java:2907)
	at android.view.View.dispatchKeyEvent(View.java:15002)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1975)
	at com.android.internal.policy.DecorView.superDispatchKeyEvent(DecorView.java:559)
	at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1889)
	at android.app.Activity.dispatchKeyEvent(Activity.java:4371)
	at androidx.core.app.ComponentActivity.superDispatchKeyEvent(Unknown Source:0)
	at androidx.tracing.Trace.dispatchKeyEvent(Unknown Source:10)
	at androidx.core.app.ComponentActivity.dispatchKeyEvent(Unknown Source:18)
	at androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(Unknown Source:22)
	at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent$androidx$appcompat$view$WindowCallbackWrapper(Unknown Source:2)
	at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(Unknown Source:19)
	at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:425)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:6945)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:6793)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6228)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6285)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6251)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:6416)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6259)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:6473)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6232)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6285)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6251)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6259)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6232)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6285)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6251)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:6449)
	at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:6646)
	at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:3692)
	at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:3200)
	at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:3191)
	at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:3669)
	at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:154)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:341)
	at android.os.Looper.loopOnce(Looper.java:169)
	at android.os.Looper.loop(Looper.java:300)
	at android.app.ActivityThread.main(ActivityThread.java:8401)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
```

### Duplicate issues

- [X] This issue has not been reported yet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by itsaky** [comment]
@IzanRodrigo I need more info on this issue.

Can you please tell me the keyboard you are using? Also, you need to provide steps to reproduce this issue.

**Comment by IzanRodrigo** [comment]
Hi there @itsaky!
Sure, I'm using the official Xiaomi Pad 6 keyboard: https://www.mi.com/global/product/xiaomi-pad-6-keyboard/
As soon as I type "`" in the code editor, the app crashes. The key associated to that character is the "Esc" key and my keyboard configuration in Android is US International.

**Comment by itsaky** [comment]
Fixed in Rosemoe/sora-editor@811331caa15a5eb957b9dd9e018110ff60d0f9f4
